File: 2 SEGUNDO/1 Cuatrimestre/Programación y Diseño Orientado a Objetos - PDOO/Practicas/javaNapakalaki/NapakalakiGame/CultistPlayer.java
package NapakalakiGame;


public class CultistPlayer extends Player{
    private static int totalCultistPlayers = 0;
    private Cultist myCultistCard ;

    public CultistPlayer(Player p ,Cultist c){
        super(p);
        this.myCultistCard = c;
    }

    @Override
    protected int getCombatLevel(){
        double val ;

        val = super.getCombatLevel() ;
        val= val * 1.20;
        val = val + (myCultistCard.getGainedLevels() * totalCultistPlayers);
        return (int)val;

    }

    @Override
    protected int getOponentLevel(Monster m){
        int res;
        res = m.getCombatLevelAgainstCultistPlayer();
        return res;
    }

    @Override
    protected boolean shouldConvert(){
      Dice dice = Dice.getInstance();
      boolean res = false;
      int number = dice.nextNumber();

      if (number == 1){
          res =true;
      }
      return res;
    }


    private Treasure giveMeATreasure(){
       Treasure res;
       int posAleatorio;
       posAleatorio = (int) (Math.random() * super.getVisibleTreasures().size());
       res = super.getVisibleTreasures().get(posAleatorio);

       return res;
    }

    private boolean canYouGiveMeATreasure(){
      boolean res = false;

      if(super.getVisibleTreasures().size()> 0){
          res = true;
      }
      return res;
    }

    public int getTotalCultistPlayers(){
        return totalCultistPlayers;
    }

    @Override
    public boolean isCultist(){
        return true;
    }
}

File: 2 SEGUNDO/1 Cuatrimestre/Programación y Diseño Orientado a Objetos - PDOO/Practicas/rubyNapakalaki/cultist_player.rb

class CultistPlayer < Player

  attr_accessor :total_cultist_players , :my_cultist_card


     def initialize(p, c)
       copia(p)
       @my_cultist_card = c

     end


    def get_combat_level

        sol = get_combat_level
        sol = sol * 1.20
        sol = sol + @my_cultist_card.get_gained_levels * @total_cultist_players

      sol

    end

    protected :get_combat_level

    def get_oponent_level(m)
       m.get_combat_level_against_cultist_player

    end

    protected :get_oponent_level

    def should_convert

      sol =false

      sol
    end

    protected :should_convert

     def giveMeATreasure

       pos_aleatorio = Random.rand(0..(@visible_treasures.length-1))
       res = @visible_treasures.at(pos_aleatorio)

       res
     end

     private :giveMeATreasure

    def canYouGiveMeATreasure

      res = false

      if @visible_treasures.length > 0

          res = true

      res
      end
    end

    private :canYouGiveMeATreasure

    def  getTotalCultistPlayers

        @total_cultist_players

    end


end
